packages/fee-estimation/README.md

```diff
@@ -0,0 +1,112 @@
+# GasEstimation Documentation
+
+## Why
+
+Gas estimation on OP-chains has both an l2 and l1 component. By default tools such as Viem, Wagmi, Ethers, and Web3.js do not support the l1 component. They will support this soon but in meantime this library can help estimate fees for transactions.
+As these tools add support for gas estimation natively this README will be updated with framework specific instructions.
+
+## High level
+
+- The l2 contract that can estimate l1Fees is called [GasPriceOracle](../contracts-bedrock/contracts/l2/GasPriceOracle.sol) contract. This library provides utils for interacting with it at a high level.
+- The GasPriceOracle is [deployed to Optimism](https://optimistic.etherscan.io/address/0x420000000000000000000000000000000000000F) and other OP chains at a predeployed address of `0x420000000000000000000000000000000000000F`
+
+For more detailed information about gas fees on Optimism's Layer 2, you can visit their [official documentation](https://community.optimism.io/docs/developers/build/transaction-fees/#the-l2-execution-fee).
+
+## Installation:
+
+```bash
+pnpm i @eth-optimism/fee-estimation
+```
+
+```bash
+npm i @eth-optimism/fee-estimation
+```
+
+```
+yarn add @eth-optimism/fee-estimation
+```
+
+## Importing the Gas Fee Estimator
+
+To use the Gas Fee Estimator in your file, import it:
+
+```typescript
+import {
+  baseFee,
+  decimals,
+  estimateFees,
+  gasPrice,
+  getL1Fee,
+  getL1GasUsed,
+  l1BaseFee,
+  overhead,
+  scalar,
+  version,
+} from '@eth-optimism/fee-estimator'
+```
+
+## Usage
+
+### Provider Options
+
+Provider options are required for most function calls. They specify the chain and RPC URL for fetching data. They can be defined as follows:
+
+```typescript
+const chainConfig = {
+  chainId: 10,
+  rpcUrl: 'https://mainnet.optimism.io',
+}
```

Comment:
will people typically have something like this chain config lying around in the same file they make those calls?

Reply:
can we make it possible or call out how to pass in stuff from an existing provider?

Reply:
Yea definitely can get this config off of an existing provider or if using wagmi/viem could import from `viem/chains`. Could even make the methods take an ethers provider or viem provider as an optional argument

Reply:
Correction. Able to offer passing in a viem client in but ethers.js provider instance makes the rpcUrl completely private https://github.com/ethers-io/ethers.js/blob/5a56fc36d7bd78ed155308ec7bd0283ece2f6dbd/src.ts/providers/provider-jsonrpc.ts#L1124.

Ethers users should have the url laying around somewhere so the documentation will tell them to create an `l2Client` with it and use that